### PR TITLE
Issue 14

### LINKED ISSUE #14: Add color option for the plot
Add matplotlib supported key colors like red, blue, black etc and hex codes

### PR DESCRIPTION
**Pull request by kvaghel1**
Resolves #14. The color option can be used for the terminal. If the color is invalid, the default color (Blue) will be used for the plot and a message will be thrown in the terminal.

Please update, if the requirements are different or resolution is required.

### CODE REVIEW

File: lib/plotutil.py [hunk 1]
```diff
@@ -32,7 +34,14 @@ def plot(func, xstart, xend, step, gui):
...
 	try:
-		plt.plot(xvals, yvals, linewidth=2.0)
+		#check if color is hex code
+		is_hex = re.search(r'^#(?:[0-9a-fA-F]{3}){1,2}$', color_name)
+		if not is_hex:
+			colors = mcolors.cnames
+			if not color_name in colors:
+				print "Color not found. Setting to default color[Blue]."
+				color_name = 'blue'
+		plt.plot(xvals, yvals, color=color_name,linewidth=2.0)
```
Comment (djmgit) [inline lib/plotutil.py]: please add a space after the **,** present after ```color_name```. Please keep the changes consistent with the code already present as this helps in enhancing code readability :)

File: plotit.py [hunk 2]
```diff
@@ -40,6 +42,9 @@ def sigint_handler(signum, frame):
 if options.stepsize:
 	stepsize = int(options.stepsize)
 
+if options.color:
+	color = str(options.color)
+
 # visualise using matplotlib
 
-plu.plot(func, xstart, xend, stepsize, False)
+plu.plot(func, xstart, xend, stepsize, color,False)
```
Comment (djmgit) [inline plotit.py]: Please add a space after the **,** present after ```color```.

File: lib/plotutil.py [hunk 3]
```diff
@@ -15,7 +17,7 @@ def create_y_values(func, xvals):
 			return
 	return yvals
 
-def plot(func, xstart, xend, step, gui):
+def plot(func, xstart, xend, step, color_name ,gui):
```
Comment (djmgit) [inline lib/plotutil.py]: Please add the space after , and not before it.

**Comment by kvaghel1** [comment]
Updated the changes requested in comments.

**Comment by djmgit** [comment]
@kvaghel1 great !! From next time please squash your commits into one, that is there should be one commit per PR. This helps in review process :)
Once again thanks for working on this :)

**Comment by Aniq55** [comment]
@kvaghel1  also make sure the PRs are descriptive and not just an issue number. you need to add the issue number in the body as , Fixes: #<issue_number>

**Pull request merged**